### PR TITLE
chore: refactor minio and kaniko to new k8s obj

### PR DESCRIPTION
**Pull request by mojtaba-esk**
This PR proposes a refactor to update Minio and Kaniko pkgs to use K8s new object instead of having clinetset & namespace separately.

### CODE REVIEW

File: pkg/minio/minio.go [hunk 1]
```diff
@@ -368,64 +367,13 @@ func (m *Minio) waitForMinio(ctx context.Context) error {
...
-}
-
-func checkServiceConnectivity(serviceEndpoint string) error {
-	conn, err := net.DialTimeout("tcp", serviceEndpoint, 2*time.Second)
-	if err != nil {
-		return ErrMinioFailedToConnect.WithParams(serviceEndpoint).Wrap(err)
-	}
-	defer conn.Close()
-	return nil // success
-}
```
Comment (smuu) [inline pkg/minio/minio.go:420]: Why did you remove these?

Comment (mojtaba-esk) [reply pkg/minio/minio.go:420]: because now we have a func in k8s package that provides `waitForService` and does the same thing.
